tensorflow_datasets/image/celeba.py

```diff
@@ -117,33 +117,39 @@ def _info(self):
     )
 
   def _split_generators(self, dl_manager):
-    extracted_dirs = dl_manager.download_and_extract({
-        "img_align_celeba": IMG_ALIGNED_DATA,
-        "list_eval_partition": EVAL_LIST,
-        "list_attr_celeba": ATTR_DATA,
-        "landmarks_celeba": LANDMARKS_DATA,
-    })
+    img_align_celeba, list_eval_partition, list_attr_celeba, landmarks_celeba = \
+      dl_manager.download([IMG_ALIGNED_DATA, EVAL_LIST, ATTR_DATA, LANDMARKS_DATA])
+
     return [
         tfds.core.SplitGenerator(
             name=tfds.Split.TRAIN,
             num_shards=10,
             gen_kwargs={
                 "file_id": 0,
-                "extracted_dirs": extracted_dirs,
+                "img_align_celeba": dl_manager.iter_archive(img_align_celeba)
```

Comment:
Not a big deal, but I feel it would be cleaner to keep the dict as it was before:
```
extracted_dirs = dl_manager.download_and_extract({
        "img_align_celeba": IMG_ALIGNED_DATA,
        "list_eval_partition": EVAL_LIST,	
         "list_attr_celeba": ATTR_DATA,	
        "landmarks_celeba": LANDMARKS_DATA,	
})

def get_iter_archives():
  return {k: dl_manager.iter_archive(arch) for k, arch in extracted_dirs}

...

gen_kwargs={
      "file_id": 0,
      "iter_archives": get_iter_archives(),
}
```

Reply:
@Conchylicultor If I am following this, I am getting this error.

```
File "/home/parths007/Github/datasets/tensorflow_datasets/image/celeba.py", line 128, in <dictcomp>
    return {k: dl_manager.iter_archive(k, arch) for k, arch in extracted_dirs}
ValueError: too many values to unpack (expected 2)
```
I am testing it by running 

`python3 -m tensorflow_datasets.scripts.download_and_prepare --datasets=celeb_a`

Here's the code after following your suggestion:
```
  def _split_generators(self, dl_manager):
    extracted_dirs = dl_manager.download_and_extract({
            "img_align_celeba": IMG_ALIGNED_DATA,
            "list_eval_partition": EVAL_LIST,	
            "list_attr_celeba": ATTR_DATA,	
            "landmarks_celeba": LANDMARKS_DATA,	
    })

    def get_iter_archives():
      return {k: dl_manager.iter_archive(arch) for k, arch in extracted_dirs}

    return [
        tfds.core.SplitGenerator(
            name=tfds.Split.TRAIN,
            num_shards=10,
            gen_kwargs={
                "file_id": 0,
                "extracted_dirs": get_iter_archives(),
            }),
        tfds.core.SplitGenerator(
            name=tfds.Split.VALIDATION,
            num_shards=4,
            gen_kwargs={
                "file_id": 1,
                "extracted_dirs": get_iter_archives(),
            }),
        tfds.core.SplitGenerator(
            name=tfds.Split.TEST,
            num_shards=4,
            gen_kwargs={
                "file_id": 2,
                "extracted_dirs": get_iter_archives(),
            })
    ]
```

Reply:
@Conchylicultor I am also confused that the purpose of this Issue was to remove the extract part and directly reading from zip but you are suggesting the same. Am I missing something?
Please correct me if I am wrong somewhere. Thanks

Reply:
My bad, I just meant keeping the dict part, but using `.download` instead of `.download_and_extract`. Thanks for the correction

```
download_files = dl_manager.download({
        "img_align_celeba": IMG_ALIGNED_DATA,
        "list_eval_partition": EVAL_LIST,	
         "list_attr_celeba": ATTR_DATA,	
        "landmarks_celeba": LANDMARKS_DATA,	
})
```
Sorry for the confusion.

Reply:
+1 - I'd also prefer passing a single map to the _generate_examples rather than 5 separate arguments

Reply:
@Conchylicultor `get_iter_archives()` is giving 
```ValueError: too many values to unpack (expected 2)```

Reply:
can you push the newest version of the code to this PR ? 

double-check if the thing that you're iterating over is a map (and not a tuple).

Reply:
If you're based on the code snippet I wrote, then I forgot a `.items()` for `extracted_dirs`

```
def get_iter_archives():
  return {
      k: dl_manager.iter_archive(arch) for k, arch in extracted_dirs.items()
  }
```